### PR TITLE
profiles: hide pals action if not installed

### PR DESCRIPTION
**Pull request by Fang-**
The conditional for this was implemented incorrectly. Additionally, the styling would not account for it.

Alongside the fix, we refactor the `!contacts` branch into a single main rendering branch, within which we check that condition to decide whether to include a component. Doing so prevents us from duplicating the now slightly more complex button rendering logic, and will make additions and modifications easier going forward.

This seems to behave correctly now, but please scrutinize closely. This is the best factoring I could come up with, but maybe it was deliberately not done like this initially.

Fixes LAND-643.

### CODE REVIEW

**Review by jamesacklin — APPROVED** [review]
Verified, does not appear on a pals-less ship.

File: ui/src/profiles/ProfileModal.tsx [hunk 1]
```diff
@@ -107,36 +71,43 @@ export default function ProfileModal() {
...
+              className="secondary-button ml-auto"
+              onClick={() => pals.addPal(ship.slice(1))}
+            >
+              Add Pal
+            </button>
+          ))}
+        <button
+          className={`secondary-button ${
+            pals.installed && ship !== window.our ? 'ml-2' : 'ml-auto'
+          }`}
```
Comment (arthyn) [inline ui/src/profiles/ProfileModal.tsx:108]: let's remove this and all the margins on the buttons and use `justify-end space-x-2` on the footer, should be simpler

**Review by arthyn — CHANGES_REQUESTED** [review]
small change, but other than that looks good to me

**Review by jamesacklin — APPROVED** [review]

**Review by arthyn — APPROVED** [review]
lgtm

**Pull request merged**